pkgs/applications/misc/premid/default.nix

```diff
@@ -0,0 +1,92 @@
+{ autoPatchelfHook, makeDesktopItem, lib, stdenv, wrapGAppsHook, fetchurl, copyDesktopItems
+, alsa-lib, at-spi2-atk, at-spi2-core, atk, cairo, cups, dbus, expat, fontconfig
+, freetype, gdk-pixbuf, glib, gtk3, libcxx, libdrm, libnotify, libpulseaudio, libuuid
+, libX11, libXScrnSaver, libXcomposite, libXcursor, libXdamage, libXext
+, libXfixes, libXi, libXrandr, libXrender, libXtst, libxcb, libxshmfence
+, mesa, nspr, nss, pango, systemd, libappindicator-gtk3, libdbusmenu
+}:
+
+stdenv.mkDerivation rec {
+  pname = "premid";
+  version = "2.3.2";
+
+  src = fetchurl {
+    url = "https://github.com/premid/Linux/releases/download/v${version}/${pname}.tar.gz";
+    sha256 = "sha256-TuID63cVZkQ2kBl2iZeuVvjRUJYBt62ppPvgffBlOXY=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    wrapGAppsHook
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    alsa-lib
+    cups
+    libdrm
+    libuuid
+    libXdamage
+    libX11
+    libXScrnSaver
+    libXtst
+    libxcb
+    libxshmfence
+    mesa
+    nss
+  ];
+
+  dontWrapGApps = true;
+  dontBuild = true;
+  dontConfigure = true;
+
+  libPath = lib.makeLibraryPath [
+    libcxx systemd libpulseaudio libdrm mesa
+    stdenv.cc.cc alsa-lib atk at-spi2-atk at-spi2-core cairo cups dbus expat fontconfig freetype
+    gdk-pixbuf glib gtk3 libnotify libX11 libXcomposite libuuid
+    libXcursor libXdamage libXext libXfixes libXi libXrandr libXrender
+    libXtst nspr nss libxcb pango systemd libXScrnSaver
+    libappindicator-gtk3 libdbusmenu
+   ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,opt/PreMiD,share/pixmaps}
+    mv * $out/opt/PreMiD
+
+    chmod +x $out/opt/PreMiD/${pname}
+    patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} \
+        $out/opt/PreMiD/${pname}
+
+    wrapProgram $out/opt/PreMiD/${pname} \
+        "''${gappsWrapperArgs[@]}" \
+        --prefix XDG_DATA_DIRS : "${gtk3}/share/gsettings-schemas/${gtk3.name}/" \
+        --prefix LD_LIBRARY_PATH : ${libPath}:$out/opt/${pname}
+
+    ln -s $out/opt/PreMiD/${pname} $out/bin/
+  '';
```

Comment:
Do we need to use opt here? maybe we can use the normal directories?

Reply:
It is to avoid collisions

Reply:
With what should thiscollide?

Reply:
collision with the libs from other electron packages like [authy](https://github.com/NixOS/nixpkgs/blob/9a43210e4f45cb3d230ca49a81bf1793adc78ac7/pkgs/applications/misc/authy/default.nix)
[discord](https://github.com/NixOS/nixpkgs/blob/9a43210e4f45cb3d230ca49a81bf1793adc78ac7/pkgs/applications/misc/authy/default.nix) also puts everything in opt to avoid this it seems